helper/schema/schema.go

```diff
@@ -1176,6 +1183,16 @@ func (m schemaMap) validateType(
 			"%q: [REMOVED] %s", k, schema.Removed))
 	}
 
+	if schema.ValidateFunc != nil {
```

Comment:
Hm, I think we should call this only if `len(es) == 0`. This way, we can assure the type is already correct. It limits the full amount of errors that can happen in a single call (you might fix the errors and more appear), but I think ValidateFunc would be way easier to write with this assumption.